### PR TITLE
Scroll More Indicator

### LINKED ISSUE #20: [BUG] Start screen has no call to action if it's scrolled off the screen
Screenshot_20200322-142945

### LINKED ISSUE #29: [BUG] First page of Your Health Checkup doesn't make it clear you need to scroll
Screenshot_20200322-151116

### LINKED ISSUE #38: [FEATURE] Scroll more indicator
Fixes #20 
Fixes #29 
Related to #34 

Theming:
Typography: Button
Icon: expand_more
Color: Same as body text on that screen

Behavior:
1. Appears when any screen is not scrolled to the end
2. Tappability is a nice to have. (Scroll a page height on tap)
3. Can exist OUTSIDE the safe area.
4. Disappears when scrolled to the end.

Scroll More

### PR DESCRIPTION
**Pull request by Hixie**
Fixes #38 
Fixes #20
Fixes #29

### CODE REVIEW

File: lib/src/ui/widgets/deferring_scroll_controller.dart [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  @override
+  void debugFillDescription(List<String> description) { }
+
+  @override
+  String toString() {
+    return '${describeIdentity(this)}($debugLabel; parented to $parent)';
+  }
+}
```
Comment (dnfield) [inline lib/src/ui/widgets/deferring_scroll_controller.dart]: new line at EOF

File: lib/src/ui/widgets/scroll_more_indicator.dart [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  void dispose() {
+    widget.controller.removeListener(_handleControllerUpdate);
+    super.dispose();
+  }
+
+  void _handleControllerUpdate() {
+    if (!widget.controller.hasClients || !widget.controller.position.haveDimensions) {
+      return;
+    }
+    bool atBottom = widget.controller.position.pixels >= widget.controller.position.maxScrollExtent;
```
Comment (dnfield) [inline lib/src/ui/widgets/scroll_more_indicator.dart]: assert these are not null?

File: lib/src/ui/widgets/scroll_more_indicator.dart [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      setState(() {
+        _atBottom = atBottom;
+      });
+    }
+  }
+
+  void _handleScrollDown() {
+    widget.controller.animateTo(
+      math.min(
+        widget.controller.position.pixels + widget.controller.position.viewportDimension,
```
Comment (dnfield) [inline lib/src/ui/widgets/scroll_more_indicator.dart]: asset not null for these and below

File: lib/src/ui/widgets/scroll_more_indicator.dart [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+          },
+        ),
+        Positioned(
+          bottom: 0.0,
+          left: 0.0,
+          right: 0.0,
+          child: IgnorePointer(
+            ignoring: _atBottom,
+            child: AnimatedOpacity(
+              opacity: _atBottom ? 0.0 : 1.0,
```
Comment (dnfield) [inline lib/src/ui/widgets/scroll_more_indicator.dart]: Since `Bob` runs an animation, we may want to drop the tree below here if opacity is 0.0

Comment (Hixie) [reply lib/src/ui/widgets/scroll_more_indicator.dart]: good call... i wonder what a good way to do that would be. might be worth us adding a TickerMode to AnimatedOpacity (not that that would immediately help here but in general)...

Comment (dnfield) [reply lib/src/ui/widgets/scroll_more_indicator.dart]: Just return a sized box when opacity is zero is probably easiest

File: lib/src/ui/widgets/scroll_more_indicator.dart (hunk 2 above)
Comment (Hixie) [reply lib/src/ui/widgets/scroll_more_indicator.dart]: added assert to widget constructor

File: lib/src/ui/widgets/scroll_more_indicator.dart (hunk 3 above)
Comment (Hixie) [reply lib/src/ui/widgets/scroll_more_indicator.dart]: added the if/return from _handleControllerUpdate here too

File: lib/src/ui/widgets/scroll_more_indicator.dart (hunk 2 above)
Comment (dnfield) [reply lib/src/ui/widgets/scroll_more_indicator.dart]: Should we still not assert that `pixels`, `maxScrollExtent` are not null?

File: lib/src/ui/widgets/scroll_more_indicator.dart (hunk 4 above)
Comment (Hixie) [reply lib/src/ui/widgets/scroll_more_indicator.dart]: hold the phone, we have an onEnd on AnimatedOpacity!

File: lib/src/ui/widgets/scroll_more_indicator.dart (hunk 2 above)
Comment (Hixie) [reply lib/src/ui/widgets/scroll_more_indicator.dart]: is it possible for them to be null when haveDimensions is true?

Comment (Hixie) [reply lib/src/ui/widgets/scroll_more_indicator.dart]: i mean, i can add asserts, certainly, but if we're going to verify every invariant the framework intends to promise, we'll be here for a long time :-)

Comment (dnfield) [reply lib/src/ui/widgets/scroll_more_indicator.dart]: Ahhh ok, I missed the haveDimensions part.  This is good then.

File: lib/src/ui/widgets/scroll_more_indicator.dart [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+              child: AnimatedOpacity(
+                opacity: _atBottom ? 0.0 : 1.0,
+                onEnd: () {
+                  setState(() {
+                    _steadyState = true;
+                  });
+                },
+                duration: const Duration(milliseconds: 100),
+                curve: Curves.easeInOut,
+                child: TickerMode(
```
Comment (dnfield) [inline lib/src/ui/widgets/scroll_more_indicator.dart:125]: @goderbauer  - does stable have the bug about nested tickermodes fixed?

**Review by ferndot — APPROVED** [review]
Looking good, thank you @Hixie!

**Pull request merged**